### PR TITLE
Improve handling cases when SAS server did not provide session state

### PR DESCRIPTION
**Pull request by YuryShkoda**
## Intent

Improve handling cases when SAS server did not provide session state.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
image
- [x] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
Mocked:
image
Server:
image
- [x] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

File: src/commands/job/execute.ts [hunk 1]
```diff
@@ -102,11 +102,28 @@ export async function execute(
       macroVars?.macroVars
     )
     .catch(async (err) => {
+      if (err?.name === 'NoSessionStatus') {
```
Comment (krishna-acondy) [inline src/commands/job/execute.ts:105]: I think an (err instanceof NoSessionStatus) check would be nicer here.

File: src/commands/job/execute.ts [hunk 2]
```diff
@@ -102,11 +102,28 @@ export async function execute(
...
+            err.logUrl,
+            100000
+          )
+
+          await saveLog(logData, logFile, jobPath, returnStatusOnly)
+        }
+
+        displayError(err)
+
+        terminateProcess(2)
```
Comment (krishna-acondy) [inline src/commands/job/execute.ts:119]: Could we not just throw the error here and let the process.exit get called in main.ts?
I think we should only have process.exit calls at the very top level.

**Review by VladislavParhomchik — DISMISSED** [review]

File: src/commands/job/execute.ts (hunk 2 above)
Comment (YuryShkoda) [reply src/commands/job/execute.ts:120]: done

File: src/commands/job/execute.ts (hunk 1 above)
Comment (YuryShkoda) [reply src/commands/job/execute.ts]: done

**Review by sabhas — APPROVED** [review]

**Pull request merged**